library/aes.c

```diff
@@ -546,14 +558,7 @@ int mbedtls_aes_setkey_enc( mbedtls_aes_context *ctx, const unsigned char *key,
     unsigned int i;
     uint32_t *RK;
 
-#if !defined(MBEDTLS_AES_ROM_TABLES)
-    if( aes_init_done == 0 )
-    {
-        aes_gen_tables();
-        aes_init_done = 1;
-
-    }
-#endif
+    MBEDTLS_AES_VALIDATE_RET( ctx != NULL && key != NULL );
```

Comment:
Why dont we split this into two? If I were a user and I get a debug trace that says: `the condition 'ctx != NULL && key != NULL' failed` it is a bit annoying because then I have to go work out whether key or ctx is the problem. Given that we have these parameter check macros, why dont we just check each branch of the && separately?